extension/chrome/elements/compose-modules/compose-input-module.ts

```diff
@@ -78,12 +79,25 @@ export class ComposeInputModule extends ViewModule<ComposeView> {
     return this.view.sendBtnModule.popover.choices.richtext;
   };
 
+  public willInputLimitBeExceeded = (textToPaste: string, targetInputField: HTMLElement, selectionLengthGetter: () => number | undefined) => {
+    const limit = 50000;
+    const toBeRemoved = selectionLengthGetter() || 0;
+    const currentLength = targetInputField.innerText.trim().length;
+    const isInputLimitExceeded = currentLength - toBeRemoved + textToPaste.length > limit;
+    return isInputLimitExceeded;
+  };
+
   private handlePaste = () => {
-    this.squire.addEventListener('willPaste', (e: WillPasteEvent) => {
+    this.squire.addEventListener('willPaste', async (e: WillPasteEvent) => {
       const div = document.createElement('div');
       div.appendChild(e.fragment);
       const html = div.innerHTML;
       const sanitized = this.isRichText() ? Xss.htmlSanitizeKeepBasicTags(html, 'IMG-KEEP') : Xss.htmlSanitizeAndStripAllTags(html, '<br>', false);
+      if (this.willInputLimitBeExceeded(sanitized, this.squire.getRoot(), () => this.squire.getSelectedText().length)) {
+        await Ui.modal.warning("The paste operation can't be completed because the resulting text size would exceed the allowed limit of 50K");
+        e.preventDefault();
```

Comment:
preventDefault() should be called before `await`

Reply:
thank you - I'll keep this in mind.